capstone/src/components/User/useSettings.jsx

```diff
@@ -0,0 +1,142 @@
+    return axios.post(`${baseURL}${preferenceBaseURL}/prioritize`, values);
+  }
+  return { res: { data: false } };
+}
+
+export default function useSettings(userType, username) {
+  const URL = defineURL(userType);
+  const isAdventurer = userType === 'adventurer';
+  const [activeIDs, setActiveIDs] = useState([]);
+  const [inactiveIDs, setInactiveIDs] = useState([]);
+  // ONLY FOR ADVENTURER
+  const [prioritize, setPrioritize] = useState(false);
+  // ONLY FOR ADVENTURER
+  function onSwitchChange() {
+    setPrioritize(!prioritize);
+  }
+
+  // get all active/inactive IDs
+  useEffect(() => {
+    if (username != null) {
+      getActiveIDs(username, URL).then((res) => setActiveIDs(res.data));
+      getInactiveIDs(username, URL).then((res) => setInactiveIDs(res.data));
+      // ONLY FOR ADVENTURER
+      getPriorization(username, userType).then((res) => setPrioritize(res.data));
+    }
+  }, [username, setActiveIDs, setInactiveIDs, setPrioritize]);
+
+  const [activeInfo, setActiveInfo] = useState([]);
+  const [inactiveInfo, setInactiveInfo] = useState([]);
+  // get information from the IDs
+  useEffect(() => {
+    getInfo(activeIDs, URL).then((res) => setActiveInfo(res.data));
+  }, [activeIDs, setActiveInfo]);
+  useEffect(() => {
+    getInfo(inactiveIDs, URL).then((res) => setInactiveInfo(res.data));
+  }, [inactiveIDs, setInactiveInfo]);
+
+  // HANDLERS
+  function onAdd(id) {
+    setActiveIDs([...activeIDs, id]);
+    setInactiveIDs(inactiveIDs.filter((inactiveID) => inactiveID !== id));
+  }
+  function onDelete(id) {
+    setInactiveIDs([...inactiveIDs, id]);
+    setActiveIDs(activeIDs.filter((activeID) => activeID !== id));
+  }
+  // Show success/error message after submission
+  const toast = useToast();
+  function submissionMessage(isSubmitted) {
+    const adventurerTitle = 'Preferences updated!';
+    const adventurerDescription = 'You can go back and find new adventures';
+    const experienceTitle = 'Information updated!!';
+    const experienceDescription = 'Get ready to outstand in this categories';
+    if (isSubmitted) {
+      return toast({
+        title: isAdventurer ? adventurerTitle : experienceTitle,
+        description: isAdventurer
+          ? adventurerDescription
+          : experienceDescription,
+        status: 'success',
+        duration: 9000,
+        isClosable: true,
+      });
+    }
+    return toast({
+      title: 'An error has ocurred',
+      description: 'Please try again later',
+      status: 'error',
+      duration: 5000,
+      isClosable: true,
+    });
+  }
+  function onSubmission() {
+    if (isAdventurer) {
+      update({ username, prioritize, activeIDs }, URL).then((res) => submissionMessage(res.data));
+    } else {
+      update({ username, activeIDs }, URL).then((res) => submissionMessage(res.data));
+    }
+  }
+  if (isAdventurer) {
+    return {
+      activeInfo,
+      inactiveInfo,
+      onSwitchChange,
+      prioritize,
+      onAdd,
```

Comment:
The fact that we're returning different fields for adventurers/non-adventurers is a clue that this hook is doing too much. It's OK for now, but if you end up adding more code to this hook, we might want to break things up.